sqlmesh/core/macros.py

```diff
@@ -1378,3 +1369,30 @@ def _coerce(
             f"Coercion of expression '{expr}' to type '{typ}' failed. Using non coerced expression at '{path}'",
         )
         return expr
+
+
+def convert_sql(v: t.Any, dialect: DialectType) -> t.Any:
+    try:
+        return _cache_convert_sql(v, dialect, v.__class__)
+    # dicts aren't hashable but are convertable
+    except TypeError:
+        return _convert_sql(v, dialect)
+
+
+def _convert_sql(v: t.Any, dialect: DialectType) -> t.Any:
+    if not isinstance(v, str):
+        try:
+            v = exp.convert(v)
+        # we use bare Exception instead of ValueError because there's
+        # a recursive error with MagicMock.
+        except Exception:
+            pass
+
+        if isinstance(v, exp.Expression):
+            v = v.sql(dialect=dialect)
```

Comment:
Should we dedent this by 1 level? The original code had it at the same level as the str instance check:

```
            if not isinstance(v, str):
                try:
                    v = exp.convert(v)
                except Exception:
                    pass

            if isinstance(v, exp.Expression):
                v = v.sql(dialect=self.dialect)
            mapping[k] = v
```

Reply:
thanks